### ISSUE TITLE
adl quit unexpectedly

### ISSUE THREAD

**Issue by jaycverg** [airsdk/Adobe-Runtime-Support#3193]
After upgrading to 50.2.5.1 from 50.2.4.1, adl no longer works.

OS: MacOS 14.4.1
Error Details:

![image 25](https://github.com/airsdk/Adobe-Runtime-Support/assets/76865/9877d21d-1c98-4c0d-9193-cdd3c3854b0c)
[error.txt](https://github.com/airsdk/Adobe-Runtime-Support/files/15060439/error.txt)

**Comment by itlancer** [comment]
The same issue: https://github.com/airsdk/Adobe-Runtime-Support/issues/3166

**Comment by apofis1969** [comment]
Same issue. 
'application descriptor file not specified
usage:
  adl ( -runtime <runtime-dir> )? ( -pubid <pubid> )? -nodebug? ( -profile PROFILE )? ( -extdir <extensions-dir> )? ( -screensize SCREEN_SIZE )? <app-desc> <root-dir>? ( -- ... )?
  adl -help

PROFILE                : mobileDevice|extendedMobileDevice|desktop|extendedDesktop|tv|extendedTV
SCREEN_SIZE            : ( <width>x<height>:<fullscreen width>x<fullscreen height> )|PREDEFINED_SCREEN_SIZE
PREDEFINED_SCREEN_SIZE : iPhone|iPhoneRetina|iPhone5Retina|iPhone6|iPhone6Plus|iPod|iPodRetina|iPod5Retina|iPad|iPadRetina|Droid|NexusOne|SamsungGalaxyS|SamsungGalaxyTab|QVGA|WQVGA|FWQVGA|HVGA|WVGA|FWVGA|1080|720|480

Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

[Process completed]
'

**Comment by apofis1969** [comment]
Hi, @pol2095 tried to do the xattr command twice. No chance. Crashes. What a mess. Using Sonoma 14.4.1

**Comment by pol2095** [comment]
Same problem, adt work correctly, Harman changed something that crashed adl...

**Comment by ajwfrost** [comment]
Yes :-( for some reason there's a crash happening within socket connections relating to the debugger. We can't actually reproduce this here, I have no idea why it's happening on some machines and not others, but with a bit of help in remote debugging from @itlancer we have got a patch available:
https://transfer.harman.com/link/7pdk2vBMTnMMFQbL3ehVOS

This contains the "Adobe AIR.framework" folder that needs to be copied over the top of the one in
`/runtimes/air/mac`

thanks

**Comment by ajwfrost** [comment]
That looks like the same issue; curious that it's appearing in 50.2.4.5, although looking at it, we had a couple of reports there of a 'hang' on start-up on iOS which is all related..  essentially a side-effect of trying to ensure we keep the behaviour the same for old applications having implemented the new TCP 'nagle' feature..
